From 6eaa71df8829933bad1d41c320ae71e2820303b3 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:24 -0500
Subject: [drm] i915: More cautious with pch fifo underruns

Message-id: <1418852380-13061-74-git-send-email-rclark@redhat.com>
Patchwork-id: 102195
O-Subject: [RHEL7 drm 73/89] drm/i915: More cautious with pch fifo underruns
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.18

commit b68362278af94e1171f5be9d4e44988601fb0439

Author:     Daniel Vetter <daniel.vetter@ffwll.ch>
AuthorDate: Mon Nov 24 17:02:45 2014 +0100
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Mon Dec 15 14:19:30 2014 -0500

    drm/i915: More cautious with pch fifo underruns

    Apparently PCH fifo underruns are tricky, we have plenty reports that
    we see the occasional underrun (especially at boot-up).

    So for a change let's see what happens when we don't re-enable pch
    fifo underrun reporting when the pipe is disabled. This means that the
    kernel can't catch pch fifo underruns when they happen (except when
    all pipes are on on the pch). But we'll still catch underruns when
    disabling the pipe again. So not a terrible reduction in test
    coverage.

    Since the DRM_ERROR is new and hence a regression plan B would be to
    revert it back to a debug output. Which would be a lot worse than this
    hack for underrun test coverage in the wild. See the referenced
    discussions for more.

    References: http://mid.gmane.org/CA+gsUGRfGe3t4NcjdeA=qXysrhLY3r4CEu7z4bjTwxi1uOfy+g@mail.gmail.com
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=85898
    References: https://bugs.freedesktop.org/show_bug.cgi?id=85898
    References: https://bugs.freedesktop.org/show_bug.cgi?id=86233
    References: https://bugs.freedesktop.org/show_bug.cgi?id=86478
    Signed-off-by: Daniel Vetter <daniel.vetter@intel.com>
    Tested-by: lu hua <huax.lu@intel.com>
    Reviewed-by: Paulo Zanoni <paulo.r.zanoni@intel.com>
    Cc: stable@vger.kernel.org
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index f06ecd3..46d11ed 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -4197,7 +4197,6 @@ static void ironlake_crtc_disable(struct drm_crtc *crtc)
   ironlake_fdi_disable(crtc);
 
   ironlake_disable_pch_transcoder(dev_priv, pipe);
-  intel_set_pch_fifo_underrun_reporting(dev, pipe, true);
 
   if (HAS_PCH_CPT(dev)) {
    /* disable TRANS_DP_CTL */
@@ -4264,7 +4263,6 @@ static void haswell_crtc_disable(struct drm_crtc *crtc)
 
  if (intel_crtc->config.has_pch_encoder) {
   lpt_disable_pch_transcoder(dev_priv);
-  intel_set_pch_fifo_underrun_reporting(dev, TRANSCODER_A, true);
   intel_ddi_fdi_disable(crtc);
  }
 
-- 
1.7.1